### PR TITLE
[Rest Server] Update restart policy to avoid stuck pending pods

### PR DESCRIPTION
**Pull request by abuccts**
Update restart policy to avoid stuck pending pods #3760.

### CODE REVIEW

File: src/rest-server/src/models/v2/job/k8s.js [hunk 1]
```diff
@@ -319,7 +319,7 @@ const generateTaskRole = (taskRole, labels, config) => {
         },
         spec: {
           privileged: false,
-          restartPolicy: 'Never',
+          restartPolicy: gangAllocation === 'true' ? 'Never' : 'OnFailure',
```
Comment (Binyang2014) [inline src/rest-server/src/models/v2/job/k8s.js]: Add comments for this? We should revert this change after we update k8s to 16.2 or above.

Comment (abuccts) [reply src/rest-server/src/models/v2/job/k8s.js]: added

**Comment by yqwang-ms** [comment]
User may not want to forever retry on failure. Could alert auto delete the pod if detected such issue? @Binyang2014

**Comment by Binyang2014** [comment]
> User may not want to forever retry on failure. Could alert auto delete the pod if detected such issue? @Binyang2014

Maybe we can use alert-manager webhook to achieve this [web-hook](https://prometheus.io/docs/alerting/configuration/#webhook_config). Create another service receive such alert and do some actions.

For better experience, admin can config this webhook's reaction when alert is received.

For now, if we want to auto delete the pod. I think we should find a way to let admin know what will happen, and admin can turn off such feature if it causes other issues.
